syntax = "proto3";

package yess;

message Event {
  int32 id = 1;
  int32 stream_id = 2;
  string type = 3;
  string payload = 4;
  int32 version = 5;
}

message Stream {
  int32 id = 1;
  string type = 2;
  int32 version = 3;
  repeated Event events = 4;
}

message Response {
  int32 status = 1;
  string msg = 2;
}

message CreateStreamReq {
  string type = 1;
}

message PushEventReq {
  int32 stream_id = 1;
  Event event = 2;
}

service YessService {
  rpc CreateStream (CreateStreamReq) returns (Response);
  rpc PushEvent (PushEventReq) returns (Response);
}
